src/movies.js

```diff
@@ -1,16 +1,126 @@
 // Iteration 1: All directors? - Get the array of all directors.
-// _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
+
+
+function getAllDirectors(movies) {                      
+    let allMovies = movies.map(function(film){          
+    // console.log('the movies are:', movies)  
+    // console.log('the movies directors  are:', movie.director)
+    return film.director                                
+     
+    })
+
+    // console.log('the movies are:', allMovies)
+    return allMovies                                   
+}
+
+
+
+//----------------------- 0 --------------------------------
+
+
+
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+
+function howManyMovies (movies){                                                
+let spielbergDramas = movies.filter(function(film){
+if (film.director === 'Steven Spielberg' && film.genre.includes('Drama')) 
+
+//console.log('las peliculas dramaticas de steven spielberg son:', film)
+return film
+
+})
+
+//console.log('las peliculas dramaticas de steven spielberg son:', spielbergDramas.length)
+return spielbergDramas.length
+
+}
+
+
+
+
+//----------------------- 0 --------------------------------
+
+
+
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+
+function ratesAverage(movies){
+
+ if (movies.length === 0) {
+    return 0
+ }
+
+let sum = movies.rate.reduce((acc, elm) => {            
+let totalSum = acc + elm
+
+},0)
+
+let average = totalSum / movies.length
+//return average
+
+return average
+
+    
+}
```

Comment:
```suggestion
function ratesAverage(movies){

 if (movies.length === 0) {
    return 0
 }

let sum = movies.reduce((acc, elm) => {            
let totalSum = acc + elm.rate
return totalSum
},0)

let average = sum / movies.length
//return average

return parseFloat(average.toFixed(2))

    
}
```

Presta atención a los cambios.

Reducimos el array por eso es movies.reduce y no movies.rate. Un array no tiene rate! Solo length.

Dentro del array de cada **elm** sacamos su **rate**.  Además tenemos que retornar el nuevo valor acumulado.

En average no es totalSum, si no sum. De esto te hubieras dado cuenta si el código estuviera bien formateado. Vas dándole a command + s para que se coloque?

Y al final lo pasamos a dos decimales con .toFixed(2). Eso nos lo convierte a string.
Lo devolvemos a number con parseFloat()